mimic/imimic.py

```diff
@@ -28,13 +28,37 @@ def resource_for_region(region, uri_prefix,
         """
 
 
+class IEndpointTemplate(Interface):
+    """
+    AN :obj:`IEndpointTemplate` provides an interface for the OS-KSCATALOG
+    Admin endpoint templates, defining the fields.
+    """
+    id_key = Attribute("Unique Service ID")
+    region_key = Attribute("Region the Endpoint services")
+    type_key = Attribute("Service type, example: object-store")
+    name_key = Attribute("Service Name, example: Cloud Files")
+    enabled_key = Attribute("Whether the Endpoint is globally available")
+    public_url = Attribute("Internet Facing URL")
+    internal_url = Attribute("Intranet Facing URL")
+    admin_url = Attribute("Service Adminsitration URL")
+    tenant_alias = Attribute("Tenant Alias to be appended to the "
+                             "endpoint URI")
+    version_id = Attribute("Version Details")
+    version_info = Attribute("URL to get the version information")
+    version_list = Attribute("URL to determine which version is hosted "
+                             "by the endpoint")
+
+
 class IExternalAPIMock(Interface):
     """
     An :obj:`IExternalAPIMock` provides an API via an entry for services
     external to Mimic in a tenant's service catalog.
     """
+    id_key = Attribute("Unique Service ID")
```

Comment:
It's part of the Keystone spec. ID is a UUID kind of thing, while Name is what the users call it (e.g Cloud Backup).